include/oc_helpers.h

```diff
@@ -94,6 +94,12 @@ typedef struct oc_mmem oc_handle_t, oc_string_t, oc_array_t, oc_string_array_t,
  */
 #define oc_free_string(ocstring) _oc_free_string(ocstring)
 
+/**
+ * @brief reset ocstring
+ *
+ */
+#define oc_set_string(ocstring, str, str_len) _oc_set_string(ocstring, str, str_len)
```

Comment:
I think this is not needed, just name the function `oc_set_string` (instead of `_oc_set_string`).

If you look at `oc_free_string` for example, the macro is there because of `OC_MEMORY_TRACE`. The call to `_oc_free_string` is different based on whether that macro is defined or not. But that is not the case for us here, so we don't need to macro override for `oc_set_string` and can define it as function directly.

Reply:
oh..  understood